extensions/ql-vscode/test/vscode-tests/no-workspace/helpers.test.ts

```diff
@@ -559,3 +569,103 @@ describe("isFolderAlreadyInWorkspace", () => {
     expect(isFolderAlreadyInWorkspace("/third/path")).toBe(false);
   });
 });
+
+describe("prepareCodeTour", () => {
+  let dir: tmp.DirResult;
+
+  beforeEach(() => {
+    dir = tmp.dirSync();
+
+    const mockWorkspaceFolders = [
+      {
+        uri: Uri.file(dir.name),
+        name: "test",
+        index: 0,
+      },
+    ] as WorkspaceFolder[];
+
+    jest
+      .spyOn(workspace, "workspaceFolders", "get")
+      .mockReturnValue(mockWorkspaceFolders);
+  });
+
+  afterEach(() => {
+    dir.removeCallback();
+  });
+
+  describe("if we're in the tour repo", () => {
+    describe("if the workspace is not already open", () => {
+      it("should open the tutorial workspace", async () => {
+        // set up directory to have a 'tutorial.code-workspace' file
+        const tutorialWorkspacePath = join(dir.name, "tutorial.code-workspace");
+        await writeFile(tutorialWorkspacePath, "{}");
+
+        // set up a .tours directory to indicate we're in the tour codespace
+        const tourDirPath = join(dir.name, ".tours");
+        await mkdir(tourDirPath);
+
+        // spy that we open the workspace file by calling the 'vscode.openFolder' command
+        const commandSpy = jest.spyOn(commands, "executeCommand");
+        commandSpy.mockImplementation(() => Promise.resolve());
+
+        await prepareCodeTour();
+
+        expect(commandSpy).toHaveBeenCalledWith(
+          "vscode.openFolder",
+          expect.anything(),
```

Comment:
You could expect this to be the (system-specific) path to the workspace file right?

Reply:
Added the specific URI we expect

Reply:
Looks like the test becomes flakey if we do this, even though we're using `Uri.parse` in both situations: https://github.com/github/vscode-codeql/actions/runs/4478429334/jobs/7871178529#step:7:231

Reply:
In the end I used `expect.objectContaining()` to just compare the path property instead of the whole URI: https://github.com/github/vscode-codeql/pull/2188/commits/8acf3e96512e2bfdff9643d596a9ac7d947de590

Reply:
[Woot! Success!](https://github.com/github/vscode-codeql/pull/2188/commits/8fd9ebf2d89ce1835040bec24177d3fd39d8edc5). 

We're able to check the path now, I just needed to compare the parsed `fsPath` off the result of `Uri.parse`, rather than the raw file path before parsing. 

----
### Aside

`toEqualPath` would have been perfect here, if I was using the comparison outside of `toHaveBeenCalledWith` instead of a nested expectation. 

I don't think I'm able to do:

```
expect(commandSpy).toHaveBeenCalledWith(
          "vscode.openFolder",
          expect(something).toEqualPath(otherThing)
)
```